src/mapgen/diamond_square/diamond_square.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "noise.h"
+#include <tuple>
+#include <list>
+
+class DiamondSquareMountain
+{
+	PseudoRandom *prandom;
+public:
+	DiamondSquareMountain(double angle, double base, double center, double rnd, PseudoRandom *prandom);
+	~DiamondSquareMountain();
+
+	void process();
+	double get_value(int x, int z);
+	void   set_value(int z, int x, double h);
+
+	double **height_map;
+	int size;
+	int n;
+
+	// random divisor
+	double big_rnd;
+
+	// random divisor for small cells
+	double small_rnd;
+	int small_rnd_size;
+
+	double base;
+};
+
+class MountainLandscape
+{
+    PseudoRandom *prandom;
+    public:
+        MountainLandscape(PseudoRandom *prandom) {this->prandom = prandom;}
+        std::list<std::tuple<DiamondSquareMountain *, std::pair<double, double>>> mountains;
```

Comment:
small bit of advice: We have a `core::vector2d<double>` type you could use here,
perhaps `core::vector2d<float>` (short `v2f`) would also suffice.